src/index.ts

```diff
@@ -233,3 +233,30 @@ class PromiseDelegate<T> {
   private _resolve: (value?: T | Thenable<T>) => void;
   private _reject: (reason?: any) => void;
 }
+
+
+/**
+ * Get global configuration data for the Jupyter application.
+ *
+ * @param name - The name of the configuration option.
+ *
+ * @returns The config value or `undefined` if not found.
+ *
+ * #### Notes
+ * For browser based applications, it is assumed that the page HTML
+ * includes a script tag with the id `jupyterConfigData` containing the
+ * configuration as valid JSON.
+ */
+export
+function getConfigOption(name: string): any {
+  if (typeof window === 'undefined') {
+    // TODO: support node here.
+    return void 0;
+  }
+  let el = document.getElementById('jupyterConfigData');
+  if (!el) {
+    return void 0;
+  }
+  let config = JSON.parse(el.textContent);
+  return (config as any)[name];
```

Comment:
`JSON.parse` returns `any`. The cast isn't needed.

We can assume the config data in the node to be static, so we only need to parse once and store the parsed result in an internal module variable.

We should also define some type-safe overloads of `getConfigOption` with the string literals for the known config options. i.e.

```
function getConfigOption(name: 'baseUrl'): string;
function getConfigOption(name: 'port'): number;
function getConfigOption(name: 'whatever'): boolean;
function getConfigOption(name: string): any {
  // implementation
}
```